vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt

```diff
@@ -108,9 +106,10 @@ class CreateRoomViewModel @AssistedInject constructor(@Assisted private val init
     }
 
     private fun initUserDomain() {
-        val displayName = session.run { getUser(myUserId) }?.toMatrixItem()?.getBestName().orEmpty()
+//        val displayName = session.run { getUser(myUserId) }?.toMatrixItem()?.getBestName().orEmpty()
 
-        setState { copy(userDomain = TchapUtils.getDomainFromDisplayName(displayName)) }
+        // Tchap: Use home server Name for user domain in create room view
+        setState { copy(userDomain = TchapUtils.getHomeServerNameFromMXIdentifier(session.myUserId)) }
```

Comment:
Don't need to comment code, we are in a specific Tchap method (I think, please check if this code exist in Element)

According to your screenshot the getHomeServerName... method is not correctly implemented. In Preprod the domain name here should be "i" and not "i.tchap.gouv.fr". Please see tchap v1 implementation of this method https://github.com/tchapgouv/tchap-android/blob/develop/vector/src/main/java/fr/gouv/tchap/util/DinsicUtils.java#L141